### PR TITLE
Align our subprocess usage with current best practices.

### PR DESCRIPTION
**Pull request by skullydazed**
## Description

This switches most of our subprocess usage to MILC's `cli.run()`. There are still a couple places that use subprocess.run which I didn't want to change just now.

I added `stdin=DEVNULL` to most commands because this fixes a bug on windows where input is no longer accepted after a subprocess is run.

## Types of Changes

- [x] Core
- [x] Bugfix

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: lib/python/qmk/os_helpers/__init__.py [hunk 1]
```diff
@@ -6,7 +6,6 @@
 import subprocess
```
Comment (fauxpark) [inline lib/python/qmk/os_helpers/__init__.py:6]:
```suggestion
from subprocess import DEVNULL
```

File: lib/python/qmk/os_helpers/__init__.py [hunk 2]
```diff
@@ -142,7 +141,7 @@ def is_executable(command):
 
     # Make sure the command can be executed
     version_arg = ESSENTIAL_BINARIES[command].get('version_arg', '--version')
-    check = run([command, version_arg], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, timeout=5, universal_newlines=True)
+    check = cli.run([command, version_arg], combined_output=True, stdin=subprocess.DEVNULL, timeout=5)
```
Comment (fauxpark) [inline lib/python/qmk/os_helpers/__init__.py:144]:
```suggestion
    check = cli.run([command, version_arg], combined_output=True, stdin=DEVNULL, timeout=5)
```

File: lib/python/qmk/keymap.py [hunk 3]
```diff
@@ -361,7 +361,7 @@ def list_keymaps(keyboard, c=True, json=True, additional_files=None, fullpath=Fa
     return sorted(names)
```
Comment (fauxpark) [inline lib/python/qmk/keymap.py:361]: `import qmk.commands` can be removed.

**Review by fauxpark — CHANGES_REQUESTED** [review]
pytest on MSYS2 fails with:
```
======================================================================
FAIL: qmk.tests.test_cli_commands.test_c2json
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:/msys64/home/fauxpark/git/qmk_firmware_develop/lib/python/qmk/tests/test_cli_commands.py", line 198, in test_c2json
    check_returncode(result)
  File "C:/msys64/home/fauxpark/git/qmk_firmware_develop/lib/python/qmk/tests/test_cli_commands.py", line 31, in check_returncode
    assert result.returncode in expected
AssertionError
```
```
qmk.tests.test_cli_commands.test_c2json ... DEBUG:MILC:Running command: ['C:/msys64/usr/bin/zsh', '-c', 'qmk c2json -kb handwired/pytest/has_template -km default keyboards/handwired/pytest/has_template/keymaps/default/keymap.c']
`C:/msys64/usr/bin/zsh -c qmk c2json -kb handwired/pytest/has_template -km default keyboards/handwired/pytest/has_template/keymaps/default/keymap.c` stdout:
<class 'TypeError'>
ERROR expected string or bytes-like object
Traceback (most recent call last):
  File "C:/msys64/mingw64/lib/python3.8/site-packages/milc/milc.py", line 467, in __call__
    return self.__call__()
  File "C:/msys64/mingw64/lib/python3.8/site-packages/milc/milc.py", line 472, in __call__
    return self._entrypoint(self)
  File "C:/msys64/home/fauxpark/git/qmk_firmware_develop/lib/python/qmk/cli/c2json.py", line 40, in c2json
    keymap_json = qmk.keymap.c2json(cli.args.keyboard, cli.args.keymap, cli.args.filename, use_cpp=cli.args.no_cpp)
  File "C:/msys64/home/fauxpark/git/qmk_firmware_develop/lib/python/qmk/keymap.py", line 545, in c2json
    keymap_json = parse_keymap_c(keymap_file, use_cpp)
  File "C:/msys64/home/fauxpark/git/qmk_firmware_develop/lib/python/qmk/keymap.py", line 519, in parse_keymap_c
    keymap_file = _c_preprocess(keymap_file)
  File "C:/msys64/home/fauxpark/git/qmk_firmware_develop/lib/python/qmk/keymap.py", line 374, in _c_preprocess
    pre_processed_keymap = cli.run(cmd, stdin=stdin)
  File "C:/msys64/mingw64/lib/python3.8/site-packages/milc/milc.py", line 125, in run
    safecmd = ' '.join(safecmd)
  File "C:/msys64/mingw64/lib/python3.8/shlex.py", line 325, in quote
    if _find_unsafe(s) is None:
TypeError: expected string or bytes-like object

returncode: 255
FAIL
```
(I'm not sure whether it's related, though)

**Review by fauxpark — APPROVED** [review]

File: lib/python/qmk/keymap.py [hunk 4]
```diff
@@ -371,7 +370,9 @@ def _c_preprocess(path, stdin=None):
     Returns:
         the stdout of the pre-processor
     """
-    pre_processed_keymap = qmk.commands.run(['cpp', path] if path else ['cpp'], stdin=stdin, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)
+    cmd = ['cpp', path] if path else ['cpp']
```
Comment (skullydazed) [inline lib/python/qmk/keymap.py:373]: @fauxpark this is the line causing `qmk pytest` to fail on msys2. If I use `str(path)` instead it works fine.

File: lib/python/qmk/keymap.py [hunk 5]
```diff
@@ -371,7 +370,9 @@ def _c_preprocess(path, stdin=None):
     Returns:
         the stdout of the pre-processor
     """
-    pre_processed_keymap = qmk.commands.run(['cpp', path] if path else ['cpp'], stdin=stdin, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)
+    cmd = ['cpp', str(path)] if path else ['cpp']
```
Comment (fauxpark) [inline lib/python/qmk/keymap.py:373]: If `path` is always an object of type `Path` then I think this should use `.as_posix()` instead, otherwise LGTM 👍

**Review by fauxpark — APPROVED** [review]

**Pull request merged**